src/app.rs

```diff
@@ -274,7 +279,14 @@ impl App {
             return Err(ServerError::UnreducedCommitment);
         }
 
-        let (tx, rx) = oneshot::channel();
+        let identity_exists = self.database.identity_exists(commitment).await?;
+        if identity_exists {
+            return Err(ServerError::DuplicateCommitment);
+        }
+
+        self.database.insert_new_identity(commitment).await?;
+
+        let (tx, _) = oneshot::channel();
         self.insert_identities_sender
```

Comment:
Also, there's now no need for this channel either. Since we put identities into the db. The `insert_identities` task can just grab them from there